### PR TITLE
Scheduled transactions for the month to show up in Account's running balance

### PR DESCRIPTION
**Pull request by joel-jeremy**
Relates to #550

### CODE REVIEW

File: packages/desktop-client/src/components/accounts/Account.js [hunk 1]
```diff
@@ -977,139 +1020,134 @@ class AccountInternal extends PureComponent {
...
-          allTransactions == null ? null : (
-            <SelectedProviderWithItems
-              name="transactions"
-              items={allTransactions}
-              fetchAllIds={this.fetchAllIds}
-              registerDispatch={dispatch => (this.dispatchSelected = dispatch)}
-            >
-              <View style={[styles.page]}>
-                <AccountHeader
+        {(allTransactions, allBalances) => (
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/Account.js:1027]: What happens if `allTransactions` is null?

File: packages/desktop-client/src/components/accounts/Account.js [hunk 2]
```diff
@@ -105,6 +112,33 @@ function AllTransactions({ account = {}, transactions, filtered, children }) {
...
+    // Reverse so we can calculate from earliest upcoming schedule.
+    let scheduledBalances = prependTransactions.map((_, index, ptArray) => {
+      let scheduledTransaction = ptArray[ptArray.length - 1 - index];
+      let amount =
+        (scheduledTransaction._inverse ? -1 : 1) * scheduledTransaction.amount;
+      return {
+        balance: (runningBalance += amount),
+        id: scheduledTransaction.id,
+      };
+    });
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/Account.js:138]: Couldn’t this use `reverse()` to be a bit more normal reading?

```suggestion
    let scheduledBalances = prependTransactions.map((scheduledTransaction) => {
      let amount =
        (scheduledTransaction._inverse ? -1 : 1) * scheduledTransaction.amount;
      return {
        balance: (runningBalance += amount),
        id: scheduledTransaction.id,
      };
    }).reverse();
```

File: packages/desktop-client/src/components/accounts/Account.js [hunk 3]
```diff
@@ -105,6 +112,33 @@ function AllTransactions({ account = {}, transactions, filtered, children }) {
     }));
   }, [schedules, accountId]);
 
+  let runningBalance = useMemo(() => {
+    if (!showBalances) {
+      return 0;
+    }
+
+    // balances is in descending order so latest order is at the start of the array.
+    return balances ? Object.values(balances)[0]?.balance ?? 0 : 0;
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/Account.js:121]: Could this be something more like `balances[transactions[0].id].balance`? Relying on the order of the object like this feels wrong to me.

Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/Account.js]: This makes much more sense. Updated.

File: packages/desktop-client/src/components/accounts/Account.js (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/Account.js]: Updated. I thought I'd avoid the extra allocation.

**Review by j-f1 — COMMENTED** [review]
Mostly looks fine to me except for the one comment about `allTransactions` being null above.

**Comment by joel-jeremy** [comment]
I have added a guard in `<AllTransactions>` to never return a null it will return an empty array instead. If I understand correctly, if `allTransactions` is an empty array, transactions table will render an "empty" (No transactions) content.

**Review by j-f1 — APPROVED** [review]
Thanks for clarifying! LGTM

**Comment by carkom** [comment]
Hi all, this PR was pulled from master on July 16th. It then never mergered in master again before it was merged itself on July 20th (not sure how this got overlooked). This means that any merges that happened from 16th to the 20th are not captured in this PR and therefore are missing from current master.

To name a few PRs that are no longer in current master (I'm sure there are many more)
#1359 
#1298
#1232

Any PRs after this one have perpetuated the issue and left out all the same PRs that this one did. Not sure the best way to fix this.

@j-f1 @MatissJanis

**Comment by carkom** [comment]
Okay, upon further investigation, it seems like only a couple lines are missing from 1232. I'll open a PR to fix them.

**Comment by carkom** [comment]
I'm finding that this PR creates a really confusing user experience with sorting and balances. 

With balance shown, sorting by any colomn does not hide the balance column but rather just blanks out all rows. Then the menu drop down doesn't show anything about hiding the balance column until you clear all sorting. Let's discuss and fix it in #1406.

**Comment by carkom** [comment]
You've changed the `showBalance={!!balances}` call in the header element but then not gone through and updated all the functions that were effected by that change (onSearch, onApplyFilter, onSort, etc.).

**Pull request merged**